plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponent.java

```diff
@@ -389,23 +391,20 @@ public void run() {
 
   private static final BucketNameValidator bucketNameValidator = new BucketNameValidator();
 
-  private String trimBucketName() {
-    String bucketName = stagingLocationInput.getText().trim();
-    if (bucketName.toLowerCase(Locale.US).startsWith("gs://")) {
-      bucketName = bucketName.substring(5);
-    }
-    return bucketName;
+  private String extractBucketNamePart() {
+    return GcsDataflowProjectClient.toGcsBucketName(stagingLocationInput.getText());
```

Comment:
You found it. I added `trim()` a while ago thinking it would only help. It doesn't seem so actually, because this was the only place `trim()` was used, and it can lead to an undesirable result due to the prefix 'gs://` being added or removed. (For example, if `  text` is entered, validation succeeds after trimming, but the options component later returns `gs://  text` without trimming.) Trimming should be consistent across everything to work correctly, so I backed it out. If a user has trailing or leading spaces, let it be an error.